Hexbin Series

mcnuttandrew opened the pull request:
![hexbin series](https://user-images.githubusercontent.com/6854312/44428646-9ded1480-a55a-11e8-951d-a98301d2ecf9.gif)

This PR introduces a hexbin series. It is a little different than previous heatmap series that we provide in that the aggregation (though we could staple similar methods into the main heatmap series) is executed on the series level rather than being provided by the user. I think this is a valuable series to add as it might be difficult for users to do otherwise, it also sits well with the uh hexagonal branding that's in the air these days.

In the current state I have not yet written docs as I am not 100% sure about the API. I'd love to hear any feedback

benshope commented:
Just chatted with @nrabinowitz because he knows his hexagons and he's going to take a look

mcnuttandrew commented:
![updated-hexes](https://user-images.githubusercontent.com/6854312/44679340-50602400-aa00-11e8-9280-fe4b48300d9d.gif)

Added docs and binning with sizing, built another example (that's pretty fun to play with tbh) and addressed @benshope 's comments. Still interested to hear comments obv
